test/modules/TestResultCoordinator/Reports/DirectMethod/DirectMethodReportGenerator.cs

```diff
@@ -82,35 +83,33 @@ public async Task<ITestResultReport> CreateReportAsync()
+                    metadata = await this.ProcessSenderAndReceiverResults(dmSenderTestResult, hasSenderResult, hasReceiverResult, networkControllerStatus, isWithinTolerancePeriod);
+                    mismatchSuccess += metadata.MismatchSuccess;
+                    mismatchFailure += metadata.MismatchFailure;
+                    hasSenderResult = metadata.HasSenderResult;
+                    hasReceiverResult = metadata.HasReceiverResult;
+
+                    if (metadata.MismatchFailure > 0 || metadata.MismatchSuccess > 0)
                     {
                         continue;
                     }
                 }
 
-                DirectMethodReportGeneratorMetadata directMethodReportGeneratorMetadataForSenderTestResults =
-                    await this.ProcessSenderTestResults(dmSenderTestResult, networkControllerStatus, isWithinTolerancePeriod, this.SenderTestResults);
-                networkOnSuccess += directMethodReportGeneratorMetadataForSenderTestResults.NetworkOnSuccess;
-                networkOffSuccess += directMethodReportGeneratorMetadataForSenderTestResults.NetworkOffSuccess;
-                networkOnToleratedSuccess += directMethodReportGeneratorMetadataForSenderTestResults.NetworkOnToleratedSuccess;
-                networkOffToleratedSuccess += directMethodReportGeneratorMetadataForSenderTestResults.NetworkOffToleratedSuccess;
-                networkOnFailure += directMethodReportGeneratorMetadataForSenderTestResults.NetworkOnFailure;
-                networkOffFailure += directMethodReportGeneratorMetadataForSenderTestResults.NetworkOffFailure;
-                hasSenderResult = directMethodReportGeneratorMetadataForSenderTestResults.HasSenderResult;
+                metadata = await this.ProcessSenderTestResults(dmSenderTestResult, networkControllerStatus, isWithinTolerancePeriod, this.SenderTestResults);
+                networkOnSuccess += metadata.NetworkOnSuccess;
+                networkOffSuccess += metadata.NetworkOffSuccess;
+                networkOnToleratedSuccess += metadata.NetworkOnToleratedSuccess;
+                networkOffToleratedSuccess += metadata.NetworkOffToleratedSuccess;
+                networkOnFailure += metadata.NetworkOnFailure;
+                networkOffFailure += metadata.NetworkOffFailure;
+                hasSenderResult = metadata.HasSenderResult;
             }
 
             while (hasReceiverResult)
             {
-                DirectMethodReportGeneratorMetadata additionalCountsAndHasResultsForMismatchFailure = await this.MismatchFailureCase();
-                mismatchFailure += additionalCountsAndHasResultsForMismatchFailure.MismatchFailure;
-                hasReceiverResult = additionalCountsAndHasResultsForMismatchFailure.HasReceiverResult;
+                metadata = await this.MismatchFailureCase();
```

Comment:
nit: rename this method; Mismatch is adjective, right? normally starts with a verb.